File: typescript/src/strings/palindromeCheck.ts
function isAlnum(c: string): boolean {
  return /[a-z0-9]/i.test(c);
}

export function palindromeCheck(s: string): boolean {
  let left = 0;
  let right = s.length - 1;
  while (left < right) {
    while (left < right && !isAlnum(s[left]!)) left++;
    while (left < right && !isAlnum(s[right]!)) right--;
    if (s[left]!.toLowerCase() !== s[right]!.toLowerCase()) return false;
    left++;
    right--;
  }
  return true;
}

File: python/src/strings/palindrome_check.py
"""Palindrome after lowercasing and keeping only alphanumeric characters."""


def palindrome_check(s: str) -> bool:
    chars = [c.lower() for c in s if c.isalnum()]
    left, right = 0, len(chars) - 1
    while left < right:
        if chars[left] != chars[right]:
            return False
        left += 1
        right -= 1
    return True
